dask/dataframe/groupby.py

```diff
@@ -1056,38 +1052,38 @@ def __init__(
         self.group_keys = group_keys
         self.obj = df
         # grouping key passed via groupby method
-        self.index = _normalize_index(df, by)
+        self.by = _normalize_by(df, by)
         self.sort = sort
 
-        if isinstance(self.index, list):
-            do_index_partition_align = all(
+        if isinstance(self.by, list):
+            do_by_partition_align = all(
```

Comment:
The name of this variable becomes a little awkward, because `by` sounds like a preposition but it's a noun in this sense. It was already a weird name, maybe just call it `aligned_to_grouper` or something?

Reply:
yeah that is a good idea.

Reply:
I simplified the logic a bit while I was at it.